### PR TITLE
*: improve the linter and fix some bugs

### LINKED ISSUE #8019: Use one linter
## Enhancement Task
We run both golangci-lint and revive linter when executing `make check`. It's better to only use one of them. Since golangci-lint is more powerful, it seems to support revive internally. we'd better only keep golangci-lint.

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #8019.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/resource_group/controller/controller.go [hunk 1]
```diff
@@ -386,8 +386,7 @@ func (c *ResourceGroupsController) Start(ctx context.Context) {
 				}
 
 			case gc := <-c.tokenBucketUpdateChan:
-				now := gc.run.now
-				go gc.handleTokenBucketUpdateEvent(c.loopCtx, now)
+				go gc.handleTokenBucketUpdateEvent(c.loopCtx)
```
Comment (rleungx) [inline client/resource_group/controller/controller.go:389]: `now` is not used, we need to confirm if it is expected.

File: client/tso_dispatcher.go [hunk 2]
```diff
@@ -797,6 +797,7 @@ func (c *tsoClient) processRequests(
 	stream tsoStream, dcLocation string, tbc *tsoBatchController,
 ) error {
 	requests := tbc.getCollectedRequests()
+	// nolint
```
Comment (rleungx) [inline client/tso_dispatcher.go:800]: defer is in the loop, not sure if it is right.

File: pkg/mcs/resourcemanager/server/config.go [hunk 3]
```diff
@@ -108,7 +108,8 @@ type ControllerConfig struct {
 }
 
 // Adjust adjusts the configuration and initializes it with the default value if necessary.
-func (rmc *ControllerConfig) Adjust(meta *configutil.ConfigMetaData) {
+// FIXME: is it expected?
+func (rmc *ControllerConfig) Adjust(_ *configutil.ConfigMetaData) {
```
Comment (rleungx) [inline pkg/mcs/resourcemanager/server/config.go:112]: `Adjust` doesn't use meta, is it expected?

File: pkg/schedule/checker/rule_checker.go [hunk 4]
```diff
@@ -347,7 +347,7 @@ func (c *RuleChecker) fixLooseMatchPeer(region *core.RegionInfo, fit *placement.
 	if region.GetLeader().GetId() != peer.GetId() && rf.Rule.Role == placement.Leader {
 		ruleCheckerFixLeaderRoleCounter.Inc()
 		if c.allowLeader(fit, peer) {
-			return operator.CreateTransferLeaderOperator("fix-leader-role", c.cluster, region, region.GetLeader().GetStoreId(), peer.GetStoreId(), []uint64{}, 0)
+			return operator.CreateTransferLeaderOperator("fix-leader-role", c.cluster, region, peer.GetStoreId(), []uint64{}, 0)
```
Comment (rleungx) [inline pkg/schedule/checker/rule_checker.go:350]: transfer leader doesn't need a source store id

File: pkg/utils/tsoutil/tso_request.go [hunk 5]
```diff
@@ -32,7 +32,7 @@ type Request interface {
 	getCount() uint32
 	// process sends request and receive response via stream.
 	// count defines the count of timestamps to retrieve.
-	process(forwardStream stream, count uint32, tsoProtoFactory ProtoFactory) (tsoResp, error)
+	process(forwardStream stream, count uint32) (tsoResp, error)
```
Comment (rleungx) [inline pkg/utils/tsoutil/tso_request.go:35]: we don't use factory in process

File: server/grpc_service.go [hunk 6]
```diff
@@ -556,7 +556,7 @@ func (s *GrpcServer) Tso(stream pdpb.PD_TsoServer) error {
 
 			if errCh == nil {
 				doneCh = make(chan struct{})
-				defer close(doneCh)
+				defer close(doneCh) // nolint
```
Comment (rleungx) [inline server/grpc_service.go:559]: `defer` here is not right.

File: pkg/utils/testutil/api_check.go [hunk 7]
```diff
@@ -100,7 +100,8 @@ func ReadGetJSONWithBody(re *require.Assertions, client *http.Client, url string
 	if err != nil {
 		return err
 	}
-	return checkResp(resp, StatusOK(re), ExtractJSON(re, data))
+	checkOpts = append(checkOpts, StatusOK(re), ExtractJSON(re, data))
```
Comment (rleungx) [inline pkg/utils/testutil/api_check.go:103]: previously this function was not right

File: pkg/schedule/splitter/region_splitter.go [hunk 8]
```diff
@@ -118,7 +119,7 @@ func (r *RegionSplitter) splitRegionsByKeys(parCtx context.Context, splitKeys []
 				r.handler.ScanRegionsByKeyRange(groupKeys, results)
 			}
 		case <-ctx.Done():
-			break
+			break outerLoop
```
Comment (rleungx) [inline pkg/schedule/splitter/region_splitter.go:122]: it won't exit the for loop

File: pkg/schedule/splitter/region_splitter_test.go [hunk 9]
```diff
@@ -76,7 +76,7 @@ func (suite *regionSplitterTestSuite) SetupSuite() {
 	suite.ctx, suite.cancel = context.WithCancel(context.Background())
 }
 
-func (suite *regionSplitterTestSuite) TearDownTest() {
+func (suite *regionSplitterTestSuite) TearDownSuite() {
```
Comment (rleungx) [inline pkg/schedule/splitter/region_splitter_test.go:79]: We need use TearDownSuite instead of TearDownTest

File: Makefile [hunk 10]
```diff
@@ -181,9 +181,6 @@ static: install-tools pre-build
 	@ gofmt -s -l -d $(PACKAGE_DIRECTORIES) 2>&1 | awk '{ print } END { if (NR > 0) { exit 1 } }'
 	@ echo "golangci-lint ..."
 	@ golangci-lint run --verbose $(PACKAGE_DIRECTORIES) --allow-parallel-runners
-	@ echo "revive ..."
-	@ revive -formatter friendly -config revive.toml $(PACKAGES)
```
Comment (JmPotato) [inline Makefile:185]: Do we need to remove `revive.toml` also?

Comment (rleungx) [reply Makefile:185]: done

File: client/resource_group/controller/controller.go (hunk 1 above)
Comment (CabinfeverB) [reply client/resource_group/controller/controller.go:389]: It's okay to remove it

File: pkg/schedule/checker/rule_checker.go (hunk 4 above)
Comment (CabinfeverB) [reply pkg/schedule/checker/rule_checker.go:350]: It isn't necessary for source.

File: pkg/mcs/resourcemanager/server/config.go (hunk 3 above)
Comment (CabinfeverB) [reply pkg/mcs/resourcemanager/server/config.go:112]: IMO, it's better use `meta` for the method `Adjust` of `RequestUnitConfig`.

**Review by JmPotato — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]

**Pull request merged**